src/Migrator.php

```diff
@@ -131,21 +134,36 @@ public function run(CapsuleInterface $capsule = null): ?MigrationInterface
                 continue;
             }
 
-            $capsule = $capsule ?? new Capsule($this->dbal->database($migration->getDatabase()));
-            $capsule->getDatabase($migration->getDatabase())->transaction(
-                static function () use ($migration, $capsule): void {
-                    $migration->withCapsule($capsule)->up();
-                }
-            );
-
-            $this->migrationTable($migration->getDatabase())->insertOne(
-                [
-                    'migration'     => $migration->getState()->getName(),
-                    'time_executed' => new \DateTime('now')
-                ]
-            );
-
-            return $migration->withState($this->resolveState($migration));
+            try {
+                $capsule = $capsule ?? new Capsule($this->dbal->database($migration->getDatabase()));
+                $capsule->getDatabase($migration->getDatabase())->transaction(
+                    static function () use ($migration, $capsule): void {
+                        $migration->withCapsule($capsule)->up();
+                    }
+                );
+
+                $this->migrationTable($migration->getDatabase())->insertOne(
+                    [
+                        'migration' => $migration->getState()->getName(),
+                        'time_executed' => new \DateTime('now')
+                    ]
+                );
+
+                return $migration->withState($this->resolveState($migration));
+            } catch (\Throwable $exception) {
+                throw new MigrationException(
+                    \sprintf(
+                        'Error in the migration (%s) occurred: %s',
+                        \sprintf(
+                            '%s (%s)',
+                            $migration->getState()->getName(),
+                            $migration->getState()->getTimeCreated()->format(self::DB_DATE_FORMAT)
+                        ),
+                        $exception->getMessage()
+                    ),
+                    $exception->getCode()
```

Comment:
you've forgot third argument, $previous. It's a good practice to set it to previous exception when you re-throw it in catch block

Reply:
added